### ISSUE TITLE
Problem with Job.processJobs

### ISSUE THREAD

**Issue by johnschult** [vsivsi/meteor-job-collection#108]
My setup is thus:

```javascript
// On Server
Mongo.Collection.prototype.constructor = Mongo.Collection;
MyJobs = new JobCollection('myJobQueue');
MyJobs.allow({
  creator: function(userId, method, params) {
    return true;
  },
  worker: function(userId, method, params) {
    return true;
  }
});

Meteor.startup(function() {
  Meteor.publish('allJobs', function() {
    return MyJobs.find({});
  });
  return MyJobs.startJobServer();
});
```

```javascript
// On Client
Mongo.Collection.prototype.constructor = Mongo.Collection;
MyJobs = new JobCollection('myJobQueue');
Meteor.subscribe('allJobs');
```
I am trying to run the standalone DDP node client as in your example and have it process jobs. I can create new jobs in the client code and can see them saved and `ready` in `MyJobs` but calls to `Job.processJobs` never finds any ready jobs. To simplify troubleshooting I am running it in the client browser console instead or messing with the remote client. The results are the same with `Job.processJobs` however using `MyJobs.processJobs` is successful.

### Finds jobs
```javascript
MyJobs.processJobs('sendMail', function(job, cb) { 
  job.done(); 
  cb();
});
```

### No jobs found
```javascript
Job.processJobs('myJobQueue', 'sendMail', function(job, cb) { 
  job.done(); 
  cb();
});
```

Am I missing something here?

**Comment by vsivsi** [comment]
Are you ever running `Job.setDDP(ddp, [collectionName])` to tell the Job object which DDP connection to use?  Also, the standalone Meteor-job package assumes the DDP npm package when it is not being run within the Meteor environment.  I'm not sure that the DDP package works correctly in a browser (as opposed to node.js) environment. I seem to remember some PRs over there to browserify it and support WebSockets, but I'm don't think those have been merged. 

Basically, using meteor-job in a browser within a Meteor app won't work as you expect because it will detect the Meteor environment and change its behavior. And using it in a browser without the Meteor client may not work because the DDP package it relies on isn't (yet) browser friendly. 

If you check you code into a git repo I can clone, and provide detailed steps to reproduce, I would be happy to take a look.

**Comment by johnschult** [comment]
I am using `Job.setDDP(ddp, [collectionName])` in the ddp client. I get what you mean by not using DDP in the browser. I will forget messing with it there then, but it still fails in my standalone client.  My code is buried in a lot of other Meteor stuff, but I will break it out into another app so I can more easily isolate it and make it available to you. Thanks for the quick reply. More news later :smile:

**Comment by johnschult** [comment]
It seems that isolating it did the trick, I think I might have had something off, but now I am not sure what it was. My project works now :grin:
